File: src/main/java/eotree/EOFloatData.java
package eotree;

/**
 * EBNF representation:
 * <code>
 * /[+-]?\d+(\.\d+)?/
 * </code>
 */
public class EOFloatData extends EOData {
    Float f;

    public EOFloatData(Float f) {
        this.f = f;
    }

    @Override
    public String generateEO(int indent) {
        return f.toString();
    }
}

File: src/main/kotlin/org/polystat/j2eo/eotree/data/EOFloatData.kt
package org.polystat.j2eo.eotree.data

/**
 * EBNF representation:
 * `
 * /[+-]?\d+(\.\d+)?/
` *
 */
class EOFloatData(
    var f: Float,
) : EOData() {
    override fun generateEO(indent: Int): String = indent(indent) + f.toString()
}
